core/src/main/scala/org/apache/spark/sql/SnappyStrategies.scala

```diff
@@ -262,15 +272,24 @@ private[sql] object JoinStrategy {
    */
   def canBroadcast(plan: LogicalPlan, conf: SQLConf): Boolean = {
     plan.statistics.isBroadcastable ||
-        plan.statistics.sizeInBytes <= conf.autoBroadcastJoinThreshold
+        plan.statistics.sizeInBytes <= conf.autoBroadcastJoinThreshold || (plan match {
+      case LogicalPlanWithHints(_, hints)
+        if hints.get(QueryHint.JoinType.toString).contains("broadcast") => true
```

Comment:
hint has to be all lowercase or you want to give the flexibility? Also for "broadcast" and other join types you want to keep some constants?